src/StorageSync/StorageSync/ServerEndpoint/NewServerEndpointCommand.cs

```diff
@@ -277,6 +281,49 @@ public override void ExecuteCmdlet()
                 string storageSyncServiceName = StorageSyncServiceName ?? ParentObject?.StorageSyncServiceName ?? parentResourceIdentifier.GetParentResourceName(StorageSyncConstants.StorageSyncServiceTypeName, 0);
                 string syncGroupName = SyncGroupName ?? ParentObject?.SyncGroupName ?? parentResourceIdentifier.ResourceName;
 
+                // Get Registered Server
+                var serverResourceIdentifier = new ResourceIdentifier(ServerResourceId);
+                StorageSyncModels.RegisteredServer registeredServer = StorageSyncClientWrapper.StorageSyncManagementClient.RegisteredServers.Get(resourceGroupName, storageSyncServiceName, serverResourceIdentifier.ResourceName);
+
+                if (registeredServer == null)
+                {
+                    throw new PSArgumentException(StorageSyncResources.MissingServerResourceIdErrorMessage);
+                }
+
+                // Handle cluster name server scenario where the rbac needs to be applied on all the nodes.
+                StorageSyncClientWrapper.VerboseLogger.Invoke($"Registered Server Auth Type : {registeredServer.ActiveAuthType} with ServerRole {registeredServer.ServerRole}");
+
+                IEnumerable<StorageSyncModels.RegisteredServer> impactedRegisteredServers;
+                if (registeredServer.ServerRole == ServerRoleType.ClusterName.ToString())
+                {
+                    impactedRegisteredServers = StorageSyncClientWrapper.StorageSyncManagementClient.RegisteredServers.ListByStorageSyncService(resourceGroupName, storageSyncServiceName);
```

Comment:
shouldn't this only be looking at servers that are in the cluster, not for the entire SSS?

Reply:
Good catch !